Mage/src/main/java/mage/cards/AdventureCardSpellImpl.java

```diff
@@ -100,18 +106,32 @@ public String getIdName() {
 
 class AdventureCardSpellAbility extends SpellAbility {
 
-    String nameFull;
+    private String nameFull;
+    private boolean finalized;
 
     public AdventureCardSpellAbility(final SpellAbility baseSpellAbility, String adventureName, CardType[] cardTypes, String costs) {
         super(baseSpellAbility);
         this.setName(cardTypes, adventureName, costs);
-        this.addEffect(ExileAdventureSpellEffect.getInstance());
         this.setCardName(adventureName);
+        this.finalized = false;
+    }
+
+    // The exile effect needs to be added last.
+    public void finalizeAdventure() {
+        if (finalized) {
+            throw new IllegalStateException("Adventures need to call finalizeAdventure() exactly once.");
```

Comment:
Try to mark developer related errors like "Wrong code usage. xxx."

Example: Wrong code usage. Adventures must call finalizeAdventure() at the end of card's setup.